Content/Items/Vitric/Weapons.PhoenixStorm.cs

```diff
@@ -0,0 +1,387 @@
+			ItemID.Sets.GamepadWholeScreenUseRange[Item.type] = true; // This lets the Player target anywhere on the whole screen while using a controller.
+			ItemID.Sets.LockOnIgnoresCollision[Item.type] = true;
+		}
+
+		public override void SetDefaults()
+		{
+			Item.damage = 40;
+			Item.knockBack = 3f;
+			Item.mana = 10;
+			Item.width = 32;
+			Item.height = 32;
+			Item.useTime = 36;
+			Item.useAnimation = 36;
+			Item.useStyle = ItemUseStyleID.Shoot;
+			Item.UseSound = SoundID.Item44;
+
+			Item.noMelee = true;
+			Item.DamageType = DamageClass.Summon;
+			Item.buffType = BuffType<PhoenixStormSummonBuff>();
+			Item.shoot = ProjectileType<PhoenixStormMinion>();
+			Item.rare = ItemRarityID.Orange;
+			Item.value = Item.sellPrice(0, 2, 0, 0);
+		}
+
+		public override void UpdateInventory(Player player)
+		{
+			stormTimer++;
+			if (player.HasMinionAttackTargetNPC)
+				stormTimer++;
+			if (stormTimer % 240 < 2)
+			{
+				List<Projectile> toSwoop = Main.projectile.Where(n => n.active && n.owner == player.whoAmI && n.ModProjectile is PhoenixStormMinion modProj && modProj.syncItem == Item).ToList();
+				toSwoop.ForEach(n => (n.ModProjectile as PhoenixStormMinion).swoopDelay = Main.rand.Next(10, 40));
+			}
+		}
+
+		public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			// This is needed so the buff that keeps your minion alive and allows you to despawn it properly applies
+			player.AddBuff(Item.buffType, 2);
+
+			// Here you can change where the minion is spawned. Most vanilla minions spawn at the cursor position.
+			position = Main.MouseWorld;
+
+			Projectile proj = Projectile.NewProjectileDirect(source, position, velocity, type, damage, knockback, player.whoAmI);
+			proj.originalDamage = damage;
+			(proj.ModProjectile as PhoenixStormMinion).syncItem = Item;
+			return false;
+		}
+
+		public override void AddRecipes()
+		{
+			Recipe recipe = CreateRecipe();
+			recipe.AddIngredient(ItemID.Book, 1);
+			recipe.AddIngredient(ItemID.HellstoneBar, 10);
+			recipe.AddIngredient(ModContent.ItemType<MagmaCore>(), 1);
+			recipe.AddTile(TileID.Anvils);
+			recipe.Register();
+		}
+	}
+
+	public class PhoenixStormMinion : ModProjectile
+	{
+		private const int AFTER_IMAGE_LENGTH = 6;
+
+		public Item syncItem;
+		public int swoopDelay = -1;
+
+		private float circleCounter = 0;
+		private bool swooping = false;
+		private float swoopSpeed = 0.1f;
+
+		private List<NPC> alreadyHit = new List<NPC>();
+
+		private List<Vector2> oldPos = new List<Vector2>();
+		private List<float> oldRot = new List<float>();
+
+		private List<Vector2> swoopTrajectory = new List<Vector2>();
+
+		private NPC target;
+
+		private Trail trail;
+		private List<Vector2> cache;
+
+		private float glowOpacity = 1;
+
+		private Player player => Main.player[Projectile.owner];
```

Comment:
Naming rule violation: Properties should be UpperCamelCase. Additionally ``Owner`` is probably a preferred name here